modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/logsdb/qa/AbstractChallengeTest.java

```diff
@@ -0,0 +1,281 @@
+
+        final AcknowledgedResponse putChallengeTemplateResponse = createTemplates(
+            getChallengeTemplateName(),
+            getChallengeDataStreamName() + "*",
+            challengeSettings,
+            challengeMappings
+        );
+        assert putChallengeTemplateResponse.isAcknowledged()
+            : "error while creating challenge template [" + getChallengeTemplateName() + "]";
+    }
+
+    private void createDataStreams() {
+        assert client().execute(CreateDataStreamAction.INSTANCE, new CreateDataStreamAction.Request(getOracleDataStreamName()))
+            .actionGet()
+            .isAcknowledged() : "error while creating data stream [" + getOracleDataStreamName() + "]";
+        assert client().execute(CreateDataStreamAction.INSTANCE, new CreateDataStreamAction.Request(getChallengeDataStreamName()))
+            .actionGet()
+            .isAcknowledged() : "error while creating data stream [" + getChallengeDataStreamName() + "]";
+    }
+
+    private AcknowledgedResponse createTemplates(
+        final String templateName,
+        final String pattern,
+        final Settings.Builder settings,
+        final XContentBuilder mappings
+    ) throws IOException {
+        final TransportPutComposableIndexTemplateAction.Request request = new TransportPutComposableIndexTemplateAction.Request(
+            templateName
+        );
+        request.indexTemplate(
+            ComposableIndexTemplate.builder()
+                .indexPatterns(List.of(pattern))
+                .template(new Template(settings.build(), new CompressedXContent(Strings.toString(mappings)), null))
+                .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate())
+                .build()
+        );
+        return client().execute(TransportPutComposableIndexTemplateAction.TYPE, request).actionGet();
+    }
+
+    private Settings.Builder createSettings(final CheckedConsumer<Settings.Builder, IOException> settingsConsumer) throws IOException {
+        final Settings.Builder settings = Settings.builder()
+            .put("index.number_of_shards", numberOfShards())
+            .put("index.number_of_replicas", numberOfReplicas());
+        settingsConsumer.accept(settings);
+        return settings;
+    }
+
+    private Settings.Builder createOracleSettings() throws IOException {
+        return createSettings(this::oracleSettings);
+    }
+
+    private Settings.Builder createChallengeSettings() throws IOException {
+        return createSettings(this::challengeSettings);
+    }
+
+    private XContentBuilder createMappings(final CheckedConsumer<XContentBuilder, IOException> builderConsumer) throws IOException {
+        final XContentBuilder builder = XContentFactory.jsonBuilder();
+        builder.startObject();
+        builder.startObject("properties");
+        builder.startObject("@timestamp");
```

Comment:
Would be nice to test with just default mapping that comes from logsdb mode as well.

Reply:
I added a `randomBoolean()` which randomly select mappings/ no mappings at all.

Reply:
And also changed the data stream names so to match the `logs-*-*` pattern.